### PR TITLE
System peer

### PR DESCRIPTION
**Pull request by folkertdev**
Select the system peer and transport its variables to the system task. They will be part of the input for clock adjustment.

### CODE REVIEW

File: ntp-proto/src/clock_select.rs [hunk 1]
```diff
@@ -23,20 +23,43 @@ const MIN_INTERSECTION_SURVIVORS: usize = 1;
...
 const MIN_CLUSTER_SURVIVORS: usize = 3;
 
+#[derive(Debug)]
+pub struct FilterAndCombine {
+    pub system_offset: NtpDuration,
+    pub system_jitter: NtpDuration,
+    pub system_peer_variables: SystemPeerVariables,
+}
+
 pub fn filter_and_combine(
```
Comment (djc) [inline ntp-proto/src/clock_select.rs:33]: Maybe rename this and move it into an `impl FilterAndCombine` block?

File: ntp-proto/src/clock_select.rs [hunk 2]
```diff
@@ -23,20 +23,43 @@ const MIN_INTERSECTION_SURVIVORS: usize = 1;
...
     );
 
-    Some(combined)
+    let filter_and_combine = FilterAndCombine {
+        system_offset: combined.system_offset,
+        system_jitter: combined.system_jitter,
+        system_peer_variables: new_system_peer,
+    };
+
+    Some(filter_and_combine)
```
Comment (djc) [inline ntp-proto/src/clock_select.rs:62]: Just do `Some(FilterAndCombine { .. })`? There's no point in having the variable binding.

Comment (folkertdev) [reply ntp-proto/src/clock_select.rs]: I think that looks weird. Rust is normally very "chain" oriented: we prefer `a.map(..).filter(..)` over `filter(map(a, ...), ...)`. But when constructors are involved it's suddenly OK to nest expressions deeply.

anyhow, not here to fight over style, it's just something about rust's formatting that I think is weird.

File: ntp-proto/src/peer.rs [hunk 3]
```diff
@@ -96,12 +96,21 @@ pub enum IgnoreReason {
     TooOld,
 }
 
+#[allow(dead_code)]
```
Comment (davidv1992) [inline ntp-proto/src/peer.rs:99]: What is dead here that requires this? Perhaps we could limit this to a single field, which also helps identify when we can remove it better.

**Review by davidv1992 — COMMENTED** [review]
Mostly looks good to me, one minor issue around an allow(dead_code) that I think we should resolve before merging.

**Pull request merged**